student-info-system/student/templates/student/registration.html

```diff
@@ -4,19 +4,32 @@
 {% load bootstrap3 %}
 
 {% block content %}
-<form action="{% url 'student:registration' %}" method="post">
-{% csrf_token %}
-  <select name="semester">
-  {% for sem in semesters %}
-    <option value="{{sem.id}}" {% if semester == sem.id %}SELECTED{% endif %}>{{ sem.name }}</option>
-  {% endfor %}
-  </select>
-
   {% if not any_sections %}
     <p>No courses offered</P>
   {% else %}
+    {% if registration_success %}
+        <p>Registration complete.</p>
+    {% endif %}
+    {% if filter %}
+      <form id="myfilt" action="" method="get" class="form form in-line" autocomplete="off">
+        {% csrf_token %}
+        {% bootstrap_form filter.form layout='inline' %}
+        <button class="btn btn-primary">Filter</button>
+        {% if has_filter %}
+          <a class-"btn btn-sucess" href="{% url 'student:registration' %}">Clear</a>
+        {% endif %}
+      </form>
+    {% endif %}
+
+  <form action="{% url 'student:registration' %}" method="post">
+    {% csrf_token %}
+    <select name="semester">
+    {% for sem in semesters %}
+      <option value="{{sem.id}}" {% if semester == sem.id %}SELECTED{% endif %}>{{ sem.name }}</option>
+    {% endfor %}
+    </select>
 
```

Comment:
I strongly suggest you look at filtered_view in sa/views, and how it interacts with the table, filter, and HTML. It supports multiple tables per page (mostly) and is kinda consistent in its use.

There are two use cases -- one for (say) students() which is a single table, and (say) student() which is some data and 1+ tables.

Reply:
I need more info, I cant find a filtered view to try to figure out the context.